public/src/frame.js

```diff
@@ -0,0 +1,51 @@
+function Frame(isFinalFrame) {
+  this.throwCount = 0;
+  this.throwScores = [];
+  //if no parameter passed in, isFinalFrame will be set to default of false
+  this.isFinalFrame = isFinalFrame || false;
+}
+
+Frame.prototype.trackThrow = function(pinsDown) {
+  if(pinsDown === 10) {
+    this.throwCount += 2;
+  }
+  this.throwScores.push(pinsDown)
+  this.throwCount ++;
+};
+
+Frame.prototype.calculateScore = function() {
+  var cumulativeScore = 0;
+  for(var i =0; i < this.throwScores.length; i++ ) {
+    cumulativeScore += this.throwScores[i];
+  }
+  return cumulativeScore;
+};
+
+Frame.prototype.isStrike = function() {
+  return (this.throwScores[0] === 10);
+};
+
+Frame.prototype.isSpare = function() {
+  return (this.throwScores[0] < 10) && (this.calculateScore() === 10);
+};
+
+Frame.prototype.addBonus = function() {
+  return (this.isSpare()) || (this.isStrike());
+};
+
+Frame.prototype.throwsPerFrame = function() {
+  if(isFinalFrame === true) {
```

Comment:
this is referencing the global variable defined in the spec, so it won't be defined in production :'-(